src/api/identity.ts

```diff
@@ -0,0 +1,101 @@
+import { ToznyAPIIdentity } from '../types/identity'
+import { validateRequestAsJSON, checkStatus } from '../utils'
+import { APIContext } from './context'
+
+type IdentityRegistrationData = {
+  name: string
+  email: string
+  first_name: string
+  last_name: string
+  public_key: string
+  signing_key: string
+}
+type RegisterIdentityData = {
+  realm_name: string
+  realm_registration_token: string
+  identity: IdentityRegistrationData
+}
+
+type RegisterIdentityResponse = {
+  identity: ToznyAPIIdentity
+  realm_broker_identity_tozny_id: string
+}
+
+type RealmInfoData = {
+  realm_name: string
+  apiUrl: string
+}
+type RealmInfoResponse = {
+  domain: string
+  name: string
+  domamain: string
+}
+export async function realmInfo({
+  realm_name,
+  apiUrl,
+}: RealmInfoData): Promise<RealmInfoResponse> {
+  let lowerCasedRealmName = realm_name.toLowerCase()
+  const response = await fetch(
+    `${apiUrl}/v1/identity/info/realm/${lowerCasedRealmName}`,
+    {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    }
+  )
+  const realmInfo = (await validateRequestAsJSON(response)) as RealmInfoResponse
+  return realmInfo
+}
+export async function registerIdentity(
```

Comment:
Can we overload this method so that we don't need to convert the registration token to a string, and can pass it in directly?

https://www.typescriptlang.org/docs/handbook/functions.html#overloads

Reply:
The method that takes whatever type the call returns `accountClient.newRegistrationToken(tokenName, permissions)` can just do the string cast and immediately call the other method.

Reply:
https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads up to date docs

Reply:
I should have cleaned up my code! the string conversion is not necessary!

Reply:
Perfect no overload needed then